### PR TITLE
build: add functional test configuration

### PR DESCRIPTION
**Pull request by rprzedzik**
Add functional test base configuration

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,11 +14,23 @@
         "guzzlehttp/guzzle": "^6.3"
     },
     "require-dev": {
+        "dama/doctrine-test-bundle": "^6.5",
+        "doctrine/doctrine-bundle": "^2.3",
+        "ergonode/authentication": "^1.0",
+        "ergonode/fixture": "dev-main as 1.1.0-dev",
+        "ergonode/product": "dev-release/1.0 as 1.0.1-dev",
```
Comment (piotrkreft) [inline composer.json:21]: it's already released. You can add exact dep

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -14,11 +14,23 @@
...
         "phpstan/phpstan": "^0.12.85",
         "phpunit/phpunit": "^9.5",
+        "rregeer/phpunit-coverage-check": "^0.3.1",
+        "qossmic/deptrac-shim": "^0.13.0",
         "slevomat/coding-standard": "^7.0",
-        "squizlabs/php_codesniffer": "^3.6"
+        "squizlabs/php_codesniffer": "^3.6",
+        "symfony/browser-kit": "^4.4",
+        "symfony/phpunit-bridge": "^4.4",
+        "symfony/security-bundle": "^4.4"
```
Comment (piotrkreft) [inline composer.json:33]: I don't think security is really required here for testing?

**Review by piotrkreft — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -14,11 +14,22 @@
         "guzzlehttp/guzzle": "^6.3"
     },
     "require-dev": {
+        "dama/doctrine-test-bundle": "^6.5",
+        "doctrine/doctrine-bundle": "^2.3",
+        "ergonode/authentication": "^1.0",
+        "ergonode/fixture": "dev-main as 1.1.0-dev",
+        "ergonode/product": "^1.0.1",
```
Comment (wfajczyk) [inline composer.json:21]: is require from relationships with channel

**Review by wfajczyk — CHANGES_REQUESTED** [review]

File: tests/Functional/ApiTest.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Ergonode\ExporterShopware6\Tests\Functional;
+
+class ApiTest extends ExporterShopware6TestCase
+{
+    public function testGetProductTypeDictionary(): void
+    {
+        $client = static::createClient();
+
+        $client->get('/api/v1/pl_PL/dictionary/product-type');
```
[the remark this answers is not in the document]
Comment (rprzedzik) [reply tests/Functional/ApiTest.php:18]: this is only simple test for check its working, should be removed after adding suitable tests

**Review by wfajczyk — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Pull request merged**